CHANGELOG.md

```diff
@@ -11,6 +11,16 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)
+- Add gRPC trace `Filter` for stats handler to `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc`. (#5196)
+
+### Changed
+
+- The gRPC trace `Filter` for interceptor is renamed to `InterceptorFilter`. (#5196)
+- The gRPC trace filter functions `Any`, `All`, `None`, `Not`, `MethodName`, `MethodPrefix`, `FullMethodName`, `ServiceName`, `ServicePrefix` and `HealthCheck` for interceptor are moved to `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc/filters/interceptor`. (#5196)
```

Comment:
I do not feel safe/comfortable with this change.

After I deprecated the interceptor, I noticed that https://pkg.go.dev/google.golang.org/grpc/stats are experimental.

@dashpole, maybe we should first ask to go-grpc team if/when then plan to stabilize the `stats` package?

Related issue: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4319

Reply:
I don't believe this PR does that.  It just moves the Filter functions to the interceptor package.

Reply:
Note that interceptors are already marked deprecated.  This PR just maintains that